instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -104,137 +104,66 @@ def _instrument(self, **kwargs):
     def _uninstrument(self, **kwargs):
         unwrap(BaseClient, "_make_api_call")
 
+    # pylint: disable=too-many-branches
     def _patched_api_call(self, original_func, instance, args, kwargs):
         if context_api.get_value("suppress_instrumentation"):
             return original_func(*args, **kwargs)
 
-        endpoint_name = deep_getattr(instance, "_endpoint._endpoint_prefix")
+        # pylint: disable=protected-access
+        service_name = instance._service_model.service_name
```

Comment:
Do we want a defensive check here in case `_service_model` is renamed or removed in future versions?

Reply:
Do you mean something like the following:

```suggestion
        service_name = instance._service_model.service_name if instance._service_model is not None else None
```

I personally don't think we have to. I'm not saying this is a very convincing reason why, but even though botocore releases minor patches almost every day, this specific property [has not changed in 6 years](https://github.com/boto/botocore/blame/develop/botocore/client.py#L628).

The tests we have will make sure the name is being correctly set, and if `botocore` _does_ update this, we can either

-  lock the version (right now it's `botocore~=1.0`) or 
- add a defensive check.

Reply:
I meant using `getattr` instead of directly accessing the attribute to avoid possible AttributeError but if you think it's not a concern with boto 1.0 then I'm fine with it.

Reply:
Sounds good! I won't add it then if everyone's alright with it.